### PR TITLE
POP3 is broken

### LINKED ISSUE #2754: Retrieving messages via pop3 fails
I upgraded my Mailu installation from 1.8 to 2.0.2. After the upgrade pop3 stopped working.

### Environment

- [x] docker compose
- [ ] kubernetes
- [ ] docker swarm

### Version

- Version: `2.0.2`

## Description
When connecting via pop3 password authentication fails due to an unexpected error. The same issue occurs on the Mailu test environment:

```bash
openssl s_client -crlf -connect test.mailu.io:110 -starttls pop3

USER admin@test.mailu.io
+OK
CAPA
+OK Capability list follows
TOP
UIDL
RESP-CODES
PIPELINING
AUTH-RESP-CODE
USER
SASL PLAIN LOGIN
.
PASS letmein
-ERR Unknown command.
+OK
closed
```

## Observed behaviour

> -ERR Unknown command.

Is returned when a valid password is provided via pop3.

## Expected behaviour
pop3 responds with an +OK when a valid password is passed.

## Workaround

I guessed the breaking change was in the front (nginx) config, as the dovecot config looked not suspicious.

So I removed the following two lines:

https://github.com/Mailu/Mailu/blob/399345e796a4e16961f6e4f218bc0a0356a82286/core/nginx/conf/nginx.conf#L415

https://github.com/Mailu/Mailu/blob/399345e796a4e16961f6e4f218bc0a0356a82286/core/nginx/conf/nginx.conf#L463

and pop3 started to work again on my server.

```text
USER mail@######.net
+OK
CAPA
+OK Capability list follows
TOP
UIDL
RESP-CODES
PIPELINING
AUTH-RESP-CODE
USER
SASL PLAIN LOGIN
.
PASS ##################
+OK Logged in.
```

### LINKED ISSUE #2757: Postfix master.cf override broken due to missing new line
### Environment

- [x] docker compose
- [ ] kubernetes
- [ ] docker swarm

### Version

- Version: `2.0`

## Description

When transferring the master.cf from /conf/ to /etc/postfix in the postfix startup script, the final new line character is stripped. The first line of the postfix.master override is appended to /etc/postfix/master.cf without this newline character as separator. Consequently, it does not take effect.

## Replication Steps
Add the following override in overrides/postfix/postfix.master.

```
showq/unix=showq unix n - n - - showq
```

## Observed behaviour
Last line of /etc/postfix/master.cf inside postfix container is:

```
postlog   unix-dgram n  -       n       -       1       postlogdshowq      unix  n       -       n       -       -       showq
```

## Expected behaviour
Last lines of /etc/postfix/master.cf inside postfix container should be:

```
postlog   unix-dgram n  -       n       -       1       postlogd
showq      unix  n       -       n       -       -       showq
```

## Solution Suggestion

Add another new line charachter to the end of the original master.cf (as it was in version 1.9) or avoid stripping the last new line character in the startup script.

### PR DESCRIPTION
**Pull request by nextgens**
## What type of PR?

bug-fix

## What does this PR do?

Add a test to show it's broken, then fix it.

### Related issue(s)
- close #2754
- close #2757

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [ ] In case of feature or enhancement: documentation updated accordingly
- [ ] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

File: .github/workflows/build_test_deploy.yml [hunk 1]
```diff
@@ -576,7 +576,8 @@ jobs:
           issue: "${{ steps.changelog.outputs.issue }}"
           changelog: "${{ steps.changelog.outputs.content }}"
         run: |
-            message="Changelog :mailbox:
+            cat << "EOT" >> release_note.md
```
Comment (Diman0) [inline .github/workflows/build_test_deploy.yml:579]:
```suggestion
            EOT="$RANDOM"
            cat << "${EOT}" >> release_note.md            
```

File: .github/workflows/build_test_deploy.yml [hunk 2]
```diff
@@ -589,7 +590,7 @@ jobs:
             The main version X.Y (e.g. 1.9) will always reflect the latest version of the branch. To update your Mailu installation simply pull the latest images \`docker compose pull && docker compose up -d\`.
 
             The pinned version X.Y.Z (e.g. 1.9.1) is not updated. It is pinned to the commit that was used for creating this release. You can use a pinned version to make sure your Mailu installation is not suddenly updated when recreating containers. The pinned version allows the user to manually update. It also allows to go back to a previous pinned version.
-            " && echo "$message" >> release_note.md
+            EOT
```
Comment (Diman0) [inline .github/workflows/build_test_deploy.yml:593]:
```suggestion
            ${EOT} 
```

**Review by Diman0 — CHANGES_REQUESTED** [review]
With the suggested changes it should be fine. The delimiter must be random to make sure an attacker cannot put the delimiter in the newsfragment and still have the ability to execute random code. 

I tested it here:
https://github.com/Diman0/Mailu_Fork/actions/runs/4644712577/jobs/8220143961
https://github.com/Diman0/Mailu_Fork/releases/tag/1.9.29

File: tests/compose/core/05_connectivity.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    #port 25 should fail
+    try:
+        print(f'Authenticating to smtps://{username}:{password}@{server}:25/')
+        with smtplib.SMTP(server) as conn:
+            conn.ehlo()
+            conn.starttls()
+            conn.ehlo()
+            conn.login(username, password)
+    except smtplib.SMTPNotSupportedError:
+        print('NOK')
```
Comment (Diman0) [inline tests/compose/core/05_connectivity.py:90]: Maybe print the text that this test is supposed to fail?

File: tests/compose/core/05_connectivity.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    try:
+        print(f'Authenticating to smtp://{username}:{password}@{server}:25/')
+        with smtplib.SMTP(server) as conn:
+            conn.ehlo()
+            conn.login(username, password)
+            print(f'Authenticating to smtp://{username}:{password}@{server}:587/ worked without STARTTLS!')
```
Comment (Diman0) [inline tests/compose/core/05_connectivity.py:96]: Shouldn't this display port 25 instead of 587?

File: tests/compose/core/05_connectivity.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            os.exit(104)
+    except smtplib.SMTPNotSupportedError:
+        print('NOK')
```
Comment (Diman0) [inline tests/compose/core/05_connectivity.py:99]: Again, we should print that NOK is fine, that authentication on port 25 should fail.

Comment (nextgens) [reply tests/compose/core/05_connectivity.py]: fixed in c008ce66

File: tests/compose/core/05_connectivity.py (hunk 4 above)
Comment (nextgens) [reply tests/compose/core/05_connectivity.py]: fixed in c008ce66

File: tests/compose/core/05_connectivity.py (hunk 3 above)
Comment (nextgens) [reply tests/compose/core/05_connectivity.py]: Fixed in c008ce66

File: .github/workflows/build_test_deploy.yml (hunk 1 above)
Comment (nextgens) [reply .github/workflows/build_test_deploy.yml]: Fixed in c008ce66

**Comment by nextgens** [comment]
bors try

File: tests/compose/core/05_connectivity.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        conn.starttls()
+        conn.ehlo()
+        conn.login(username, password)
+        print('OK')
+    try:
+        print(f'Authenticating to smtp://{username}:{password}@{server}:587/')
+        with smtplib.SMTP(server, 587) as conn:
+            conn.ehlo()
+            conn.login(username, password)
+            print(f'Authenticating to smtp://{username}:{password}@{server}:587/ worked!')
```
Comment (Diman0) [inline tests/compose/core/05_connectivity.py:79]: I think the print line should mention it worked without starttls
the following line should have an non-zero exit code isn't it? This test should fail after all
```suggestion
            print(f'Authenticating to smtp://{username}:{password}@{server}:587/ worked without STARTTLS!')
            os.exit(104)
```

File: tests/compose/core/05_connectivity.py [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        print('NOK - expected')
+    #port 25 should fail
+    try:
+        print(f'Authenticating to smtps://{username}:{password}@{server}:25/')
+        with smtplib.SMTP(server) as conn:
+            conn.ehlo()
+            conn.starttls()
+            conn.ehlo()
+            conn.login(username, password)
+            print(f'Authenticating to smtps://{username}:{password}@{server}:25/ worked!')
```
Comment (Diman0) [inline tests/compose/core/05_connectivity.py:90]:
```suggestion
            print(f'Authenticating to smtps://{username}:{password}@{server}:25/ worked!')
            os.exit(105)
```
When the test succeeds, it actually failed and we need a non-zero return code

File: tests/compose/core/05_connectivity.py [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    except smtplib.SMTPNotSupportedError:
+        print('NOK - expected')
+    try:
+        print(f'Authenticating to smtp://{username}:{password}@{server}:25/')
+        with smtplib.SMTP(server) as conn:
+            conn.ehlo()
+            conn.login(username, password)
+            print(f'Authenticating to smtp://{username}:{password}@{server}:25/ worked without STARTTLS!')
+            os.exit(104)
```
Comment (Diman0) [inline tests/compose/core/05_connectivity.py:99]:
```suggestion
            os.exit(106)
```
due to the above changes this number must be bumped to make it unique.

**Review by Diman0 — CHANGES_REQUESTED** [review]
I think we are almost there. There are only a couple of tests that miss a non-zero exit code in case the tests succeed (while they should always fail)

**Comment by Diman0** [comment]
bors try

**Review by Diman0 — APPROVED** [review]
BB8 Thumbs Up

**Pull request merged**